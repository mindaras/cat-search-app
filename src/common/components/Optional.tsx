interface Props {
  if: boolean;
  children: any;
}

const Optional: React.FC<Props> = (props) => {
  if (props.if) {
    return props.children;
  }

  return null;
};

export { Optional };
