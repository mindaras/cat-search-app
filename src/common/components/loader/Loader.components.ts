import styled from "styled-components";

const Container = styled.div`
  @keyframes rotate {
    0% {
      transform: rotate(0deg) scale(0.8);
    }
    50% {
      transform: rotate(360deg) scale(1.2);
    }
    100% {
      transform: rotate(720deg) scale(0.8);
    }
  }

  @keyframes ball1 {
    0% {
      box-shadow: 30px 0 0 ${(p) => p.theme.colors.gray.medium};
    }
    50% {
      box-shadow: 0 0 0 ${(p) => p.theme.colors.gray.medium};
      margin-bottom: 0;
      transform: translate(15px, 15px);
    }
    100% {
      box-shadow: 30px 0 0 ${(p) => p.theme.colors.gray.medium};
      margin-bottom: 10px;
    }
  }

  @keyframes ball2 {
    0% {
      box-shadow: 30px 0 0 ${(p) => p.theme.colors.gray.medium};
    }
    50% {
      box-shadow: 0 0 0 ${(p) => p.theme.colors.gray.medium};
      margin-top: -20px;
      transform: translate(15px, 15px);
    }
    100% {
      box-shadow: 30px 0 0 ${(p) => p.theme.colors.gray.medium};
      margin-top: 0;
    }
  }

  animation: rotate 2s infinite;
  height: 50px;
  width: 50px;

  :before,
  :after {
    border-radius: 50%;
    content: "";
    display: block;
    height: 20px;
    width: 20px;
  }

  :before {
    animation: ball1 2s infinite;
    background-color: ${(p) => p.theme.colors.gray.medium};
    box-shadow: 30px 0 0 ${(p) => p.theme.colors.gray.medium};
    margin-bottom: 10px;
  }

  :after {
    animation: ball2 2s infinite;
    background-color: ${(p) => p.theme.colors.gray.medium};
    box-shadow: 30px 0 0 ${(p) => p.theme.colors.gray.medium};
  }
`;

export { Container };
