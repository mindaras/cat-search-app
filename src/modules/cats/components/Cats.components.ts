import styled from "styled-components";

const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: calc(
    100vh - ${(p) => p.theme.sizes.search.height} -
      ${(p) => p.theme.sizes.pagination.height}
  );
`;

export { LoaderContainer };
