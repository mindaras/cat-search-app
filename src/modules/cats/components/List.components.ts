import styled from "styled-components";

const Cat = styled.img`
  width: 25%;
  height: calc(100% / 3);
  object-fit: cover;
  border: 4px solid #fff;
  border-radius: 30px;

  ${(p) => p.theme.media.medium} {
    width: 33.33%;
    height: 30vw;
  }

  ${(p) => p.theme.media.small} {
    width: 100%;
    height: 100vw;
  }
`;

interface ContainerProps {
  single: boolean;
}

const Container = styled.div<ContainerProps>`
  position: relative;
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  height: calc(
    100vh - ${(p) => p.theme.sizes.search.height} -
      ${(p) => p.theme.sizes.pagination.height}
  );

  ${(p) => {
    if (p.single) {
      return `
            justify-content: center;
            align-items: center;

            ${Cat} {
                width: 40vw;
                height: 40vw;

                ${p.theme.media.medium} {
                    width: 50vw;
                    height: 50vw;
                }

                ${p.theme.media.small} {
                    width: 80vw;
                    height: 80vw;
                }
            }
        `;
    }
  }};

  ${(p) => p.theme.media.medium} {
    height: auto;
  }
`;

const NoDataText = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 24px;
`;

const NoDataContainer = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: calc(100vh - ${(p) => p.theme.sizes.search.height});

  ${Cat} {
    opacity: 0.1;
  }

  ${(p) => p.theme.media.small} {
    height: auto;
  }
`;

export { Cat, Container, NoDataContainer, NoDataText };
