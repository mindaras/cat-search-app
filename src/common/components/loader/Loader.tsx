import { Container } from "./Loader.components";

const Loader = () => {
  return <Container></Container>;
};

export { Loader };
