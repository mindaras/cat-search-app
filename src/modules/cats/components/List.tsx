import { Optional } from "@common/components/Optional";
import * as Types from "../common/models";
import { Cat, Container, NoDataContainer, NoDataText } from "./List.components";
import pawImage from "@common/images/paw.jpeg";

interface Props {
  cats: Types.Cat[];
}

const List: React.FC<Props> = ({ cats }) => (
  <Container single={cats.length < 2}>
    <Optional if={!!cats.length}>
      {cats.map((cat) => (
        <Cat key={cat.id} src={cat.url} alt="Cat" />
      ))}
    </Optional>
    <Optional if={!cats.length}>
      <NoDataContainer>
        <Cat src={pawImage} alt="Paw" />
        <NoDataText>No data</NoDataText>
      </NoDataContainer>
    </Optional>
  </Container>
);

export { List };
