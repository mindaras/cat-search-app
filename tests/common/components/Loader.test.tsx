import renderer from "react-test-renderer";
import { Container } from "@testUtils/components/Container";
import { Loader } from "@common/components/loader/Loader";

describe("Loader", () => {
  it("matches a snapshot", () => {
    const component = renderer.create(
      <Container>
        <Loader />
      </Container>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
