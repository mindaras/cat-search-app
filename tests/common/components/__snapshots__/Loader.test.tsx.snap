// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches a snapshot 1`] = `
<div
  className="sc-bcXHqe bheUzg"
/>
`;
